client-react/src/features/todos/components/TodoModal/TodoModal.tsx

```diff
@@ -0,0 +1,110 @@
+import { Button, Box, Modal, TextField, Typography } from "@mui/material"
+import { useForm } from "react-hook-form";
+import { useAppDispatch, useAppSelector } from "../../../../app/hooks";
+import { TodosCollectionActions } from "../../../../store/actions";
+import { TodoInputs } from "../../models/todoInputs.model";
+import { TodoModalsActions } from "../../store/actions";
+import { v4 as uuidv4 } from 'uuid';
+import styles from "./TodoModal.module.scss"
+import { TodoSelectors } from "../../store/selector";
+import { useEffect } from "react";
+
+
+
+interface TodoModalProps {
+  isOpen: boolean,
+}
+
+const style = {
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  p: 4,
+};
+
+export const TodoModal = ({isOpen}: TodoModalProps) => {
+  const todoBeingEdited = useAppSelector(TodoSelectors.todoBeingEdited)
+  const { register, handleSubmit, reset, formState: { errors } } = useForm<TodoInputs>();
+  const dispatch = useAppDispatch()
+
+  const clearFormState: TodoInputs = {
```

Comment:
This is a constant object that doesn't needs to be reinitialized each time our component is rerendered. Move this outside of the component.

Reply:
fixed